numpy/core/src/multiarray/cblasfuncs.c

```diff
@@ -457,7 +457,7 @@ cblas_matrixproduct(int typenum, PyArrayObject *ap1, PyArrayObject *ap2,
             Py_DECREF(ap1);
             Py_DECREF(ap2);
             Py_DECREF(out_buf);
-            return PyArray_Return(result);
+            return PyArray_Return(out_buf);
```

Comment:
What's this change about? It looks pretty dubious...

(This looks like it might be related to a recent fix to `dot` on master -- maybe the `+` line here is what master used to have, and somehow in your rebasing you ended up reverting it?)

Reply:
Good catch